modules/ui/src/main/scala/lucuma/ui/optics/ChangeAuditor.scala

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.optics
+
+import cats.data.Validated._
+import cats.syntax.all._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.api.{ Validate => RefinedValidate }
+import lucuma.core.math._
+import mouse.all._
+
+sealed trait AuditResult[A] extends Product with Serializable
+object AuditResult {
+  case class Reject[A]() extends AuditResult[A]
+  case class Accept[A]() extends AuditResult[A]
+  case class NewString[A](newS: String, cursorOffset: Int) extends AuditResult[A]
+
+  def reject[A]: AuditResult[A] = Reject()
+  def accept[A]: AuditResult[A] = Accept()
+  def newString[A](newS: String, cursorOffset: Int = 0): AuditResult[A] =
+    NewString(newS, cursorOffset)
+}
+
+sealed trait FilterMode extends Product with Serializable
+object FilterMode {
```

Comment:
Should we make this an `Enumerated`?

Reply:
We can. Is there an advantage to doing so?